app/models/spree/flow_io_order_decorator.rb

```diff
@@ -23,6 +23,13 @@ def flow_order
       flow_data&.[]('order')
     end
 
+    def flow_order?
+      payment = payments.completed.first
+      return payment.payment_method.type == 'Spree::Gateway::FlowIo' if payment
+
+      flow_data.present?
```

Comment:
`flow_data` is not enough to check - better use `flow_order` method:

```suggestion
      flow_order.present?
```